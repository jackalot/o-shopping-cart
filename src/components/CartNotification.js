import React from "react";
import "../css-component-files/CartNotification.css"
function CartNotification (props) {
    return <div onClick={props.handler} className="CartNotification-Container">
        <div className="CartNotification-Amount">
            Cart: {props.amount}
        </div>
    </div>
}
export default CartNotification;