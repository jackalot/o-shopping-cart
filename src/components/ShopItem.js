import React, { useState } from "react";
import '../css-component-files/ShopItem.css'
function ShopItem (props) {
    const [quantity, setQuantity] = useState(0);
    function changeQuantity(newQuantity) {
        setQuantity(newQuantity);
    }
    return <div className="ShopItem-Container">
        <div className="ShopItem-ItemName">
       {props.itemName}
        </div>
        <div className="ShopItem-Price">
        Price: {props.itemPrice}
        </div>
        <label onChange={changeQuantity} for={props.itemName + "quantity"}>Quantity</label>
        <input className="QuantityInput" name={props.itemName + "quantity"} type="number"></input>
        <button className="AddToCartBtn" onClick={()=> props.handler(props.itemName, props.itemPrice)}>Add To Cart!</button>
    </div>
}
export default ShopItem;