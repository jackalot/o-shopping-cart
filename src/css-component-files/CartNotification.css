.CartNotification-Container {
    width: fit-content
}
.CartNotification-Amount {
    background-color: #FC2E20;
    font-size: 30px;
    border-color: #FD7F20;
    border-style: solid;
    border-radius: 20px;
    border-width: 10px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    padding-top: 10px;
    padding-bottom: 10px;
    padding-right: 20px;
    padding-left: 20px;
}