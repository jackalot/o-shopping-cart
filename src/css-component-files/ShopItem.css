.ShopItem-Container {
    background-color: #FDB750;
    width: fit-content;
    display: flex;
    flex-direction: column;
    justify-content: center;
}
.ShopItem-ItemName {
    font-size: 40px;
    padding-left: 20px;
    padding-right: 20px;
}
.ShopItem-Price {
    font-size: 20px;
    padding-left: 20px;
    padding-right: 20px;
}
.QuantityInput {
    padding-left: 20px;
    padding-right: 20px;
}
.AddToCartBtn {
    border-color: #010100;
}