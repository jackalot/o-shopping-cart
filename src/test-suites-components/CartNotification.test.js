import React from "react";
import "@testing-library/jest-dom"
import userEvent from "@testing-library/user-event"; 
import { getByDisplayValue, getByLabelText, getByRole, getByText, render } from "@testing-library/react";
import ShopItem from "../components/ShopItem";
import CartNotification from "../components/CartNotification";

describe("CartNotification Component", () => {
    it("creates a snapshot for the whole component", () => {
        const { container } = render(<CartNotification
             amount = {3}
             ></CartNotification>)
        expect(container).toMatchSnapshot();
    })
    it("The 'Cart' Text Displays the number 4", () => {
        const { container }  = render(<CartNotification
            amount = {4}
            ></CartNotification>)
        const result = getByText(container, "Cart: 4")
        expect(result.textContent).toBe("Cart: 4");
    })
    it("The 'Cart' Text Displays the number 25", () => {
        const { container }  = render(<CartNotification
            amount = {25}
            ></CartNotification>)
        const result = getByText(container, "Cart: 25")
        expect(result.textContent).toBe("Cart: 25");
    })
    it("Cart Notification calls handleOpenCart when clicked", () => {
        let cartOpened = false;
        function handleOpenCart(){
            cartOpened = true;
        }
        const { container } = render(<CartNotification handler={handleOpenCart} amount={2}></CartNotification>)
    })
})