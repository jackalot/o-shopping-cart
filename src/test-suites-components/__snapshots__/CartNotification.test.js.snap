// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartNotification Component creates a snapshot for the whole component 1`] = `
<div>
  <div
    class="CartNotification-Container"
  >
    <div
      class="CartNotification-Amount"
    >
      Cart: 
      3
    </div>
  </div>
</div>
`;
